### ISSUE TITLE
False positive: "Cannot borrow immutable local variable as mutable"

### ISSUE THREAD

**Issue by pwlodarczyk92** [intellij-rust/intellij-rust#2865]
## Environment

* **Intellij-Rust plugin version: 0.2.0.2422-182-nightly**
* **Rust toolchain version: rustc 1.30.0-nightly (cb6d2dfa8 2018-09-16)**
* **IDE name and version: IntelliJ IDEA Ultimate 2018.2.3**
* **Operating system: Ubuntu 16.04 LTS**

## Problem description

Mutably borrowing field from immutably bound mutable struct reference is incorrectly reported as an error.

## Steps to reproduce
```rust
fn main() {
    let mut x = (0, 0);
    let y = &mut x;
    let _z = &mut y.0;
}
```
Cannot borrow immutable local variable `y.0` as mutable
